lib/sshutils/up.go

```diff
@@ -187,19 +206,54 @@ func (u *Upstream) PipeShell(rw io.ReadWriter, req *PTYReqParams) error {
 		u.session.SendRequest(PTYReq, false, ssh.Marshal(*req))
 	}
 
+	// getPrefix protects u.prefix with a mutex
+	getPrefix := func() []byte {
+		u.Lock()
+		defer u.Unlock()
+		return u.prefix
+	}
+
+	// copyOutput works exactly like io.Copy() but it does two additional things:
+	// It appends 'prefix' in front of every write (used to send screensize back to
+	// the web client in real time (it MUST know the screen size ahead of every write)
+	copyOutput := func(w io.Writer, r io.Reader) (err error) {
```

Comment:
let's make this a separate function, hard to read like closure
let's document why are we forced to use limited buffer here